Add uniform error and fatal message handling functions.

qbarnes opened the pull request:
Convert cw2dmk.c to using the new functions.

qbarnes committed:
Add uniform error and fatal message handling functions.

Convert cw2dmk.c to using the new functions.

TimothyPMann commented:
This is a good cleanup. Let's use the gcc "attribute format" to declare these as printf-like so that we get checking.

qbarnes committed:
Integrate review feedback.  Add printf-like attributes to new
functions when compiling with GCC.  Also add one for the previously
existing msg() function.  That turned up a problem when msg() is
invoked which is also corrected.

qbarnes commented:
Excellent idea!

I added the attributes for the new functions, but also thought to add them for msg() too, which turned up a small problem, so I fixed that.

TimothyPMann commented:
Nice. That was a funny little bug in msg usage. Must have been a copy/paste/incomplete-edit.
